apiserver/facades/agent/caasapplication/application.go

```diff
@@ -40,13 +41,17 @@ type Facade struct {
 func NewStateFacade(ctx facade.Context) (*Facade, error) {
 	authorizer := ctx.Auth()
 	resources := ctx.Resources()
-	return NewFacade(resources, authorizer, &stateShim{ctx.State()}, ctx.StatePool().Clock())
+	return NewFacade(resources, authorizer,
+		&stateShim{ctx.StatePool().SystemState()},
```

Comment:
This doesn't appear to required the shim wrapper. If not, including it hinders clarity somewhat...